src/pybind/mgr/dashboard/frontend/src/app/shared/api/grafana.service.ts

```diff
@@ -0,0 +1,19 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+import { ApiModule } from './api.module';
+
+@Injectable({
+  providedIn: ApiModule
+})
+export class GrafanaService {
+  port: number;
+  protocol: string;
+  host: string;
+
+  constructor(private http: HttpClient) {}
+
+  getUrl() {
```

Comment:
Since all settings declared in the `settings.py` are accessible from the `/api/settings` endpoint, what about renaming this service to `SettingsService`, and this method to `getGrafanaApiUrl()`?

IMO this will help to add access to other settings in the future.